radio/src/pulses/pulses_avr.cpp

```diff
@@ -94,135 +93,140 @@ uint8_t *pulses2MHzWPtr = pulses2MHz;
+      OCR1B = OCR1A;
+      if (g_model.pulsePol) {
+        TCCR1A = (TCCR1A | (1<<COM1B1)) & ~(1<<COM1B0); // Set idle level.
+      }
+      else {
+        TCCR1A |= 3<<COM1B0;
+      }
+      TCCR1C = 1<<FOC1B; // Strobe FOC1B.
+      TCCR1A = (TCCR1A | (1<<COM1B0)) & ~(1<<COM1B1); // Toggle OC1B on next match.
+#endif
     }
-    setupPulses(); // does not sei() for setupPulsesPPM
+    setupPulses(); // Does not sei() for setupPulsesPPM.
     heartbeat |= HEART_TIMER_PULSES;
     return;
   }
 
-#if !defined(CPUM2560)
-  // Original bitbang for PPM
   if (s_current_protocol[0] != PROTO_NONE) {
+#if !defined(CPUM2560)
+    // Original Bit-bang for PPM.
     if (g_ppmPulsePolarity) {
-      PORTB |=  (1<<OUT_B_PPM); // GCC optimisation should result in a single SBI instruction
+      PORTB |= (1<<OUT_B_PPM); // GCC optimisation should result in a single SBI instruction
       g_ppmPulsePolarity = 0;
     }
     else {
       PORTB &= ~(1<<OUT_B_PPM);
       g_ppmPulsePolarity = 1;
     }
-  }
 #else // defined(CPUM2560)
-  // CPUM2560 zero jitter hardware toggled PPM_out
-  OCR1B = *((uint16_t*)pulses2MHzRPtr); // duplicate capture (Timer1 in CTC mode, so restricted to OCR1A for int vector)
-    
-  // Toggle bit: Can't read PPM_OUT I/O pin when OC1B is connected (on the ATmega2560 -- can on ATmega64A!)
-  // so need to use pusePol register to keep track of PPM_out polarity.
-  if (s_current_protocol[0] != PROTO_NONE) {
-    if (g_ppmPulsePolarity) {
-      TCCR1A = (3<<COM1B0); // SET the state of PB6(OC1B)/PPM_out on next TCNT1==OCR1B
-      g_ppmPulsePolarity = 0;
+    // CPUM2560 hardware toggled PPM out.
+    if (*(uint16_t*)(pulses2MHzRPtr + sizeof(uint16_t)) == 0) {
+      // Look one step ahead to see if we are currently the "rest" period.
+      OCR1B = 0xffff; // Prevent next compare match hence toggle.
     }
     else {
-      TCCR1A = (2<<COM1B0); // CLEAR the state of PB6(OC1B)/PPM_out on next TCNT1==OCR1B
-      g_ppmPulsePolarity = 1;
+      OCR1B = *((uint16_t*) pulses2MHzRPtr);
     }
-  }
 #endif
+  }
+
+  OCR1A = *((uint16_t*) pulses2MHzRPtr); // Schedule next Timer1 interrupt vector (to this function).
+  pulses2MHzRPtr += sizeof(uint16_t); // Non PPM protocols use uint8_t pulse buffer.
 
-  OCR1A = *((uint16_t*)pulses2MHzRPtr); // Schedule next Timer1 interrupt vector (to this function)
-  pulses2MHzRPtr += sizeof(uint16_t); // non PPM protocols use uint8_t pulse buffer
-  
   if (dt > g_tmr1Latency_max) g_tmr1Latency_max = dt;
   if (dt < g_tmr1Latency_min) g_tmr1Latency_min = dt;
 }
 
 void setupPulsesPPM(uint8_t proto)
 {
-  // Total frame length is a fixed 22.5msec (more than 9 channels is non-standard and requires this to be extended.)
-  // Each channel's pulse is 0.7 to 1.7ms long, with a 0.3ms stop tail, making each compelte cycle 1 to 2ms.
+    // Total frame length is a fixed 22.5msec (more than 9 channels is non-standard and requires this to be extended.)
```

Comment:
Remove two spaces indentation on this and all following lines that have the same indentation added.